tests/connector/tests/main_test.go

```diff
@@ -40,20 +43,40 @@ var (
 )
 
 func TestMain(m *testing.M) {
-	logrus.Info("Starting Connector Test")
+	log.Info("Starting Connector Test")
 	cfg = config.ConnectorTestConfig{}
 	config.ReadConfig(&cfg)
 	ctx = context.Background()
 
+	var dexToken string
+
+	tokenConfig := server.Config{}
+	err := envconfig.InitWithPrefix(&tokenConfig, "APP")
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	log.Info("Get Dex id_token")
+	if tokenConfig.IsWithToken {
```

Comment:
I see this block on multiple places:
```
if tokenConfig.IsWithToken {
		tokenConfig.Log = log.Infof
		ts := server.New(&tokenConfig)
		dexToken = server.WaitForToken(ts)
	} else {
		dexToken, err = idtokenprovider.GetDexToken()
		if err != nil {
			log.Fatal(errors.Wrap(err, "while getting dex token"))
		}
	}
```

I was wondering if we could hide it behind some sort of abstraction. One idea maybe is to have `clients.NewStaticUserClient` & `clients.NewStaticUserClientWithToken` and the logic hidden behind those two. But not sure we want to hide the server opening inside the `clients` package abstraction.